service/resource/endpoint/resource.go

```diff
@@ -79,6 +97,51 @@ func getAnnotations(pod apiv1.Pod, ipAnnotationName string, serviceAnnotationNam
 	return ipAnnotationValue, serviceAnnotationValue, nil
 }
 
+func isEmptyEndpoint(endpoint apiv1.Endpoints) bool {
+	for _, subset := range endpoint.Subsets {
+		if len(subset.Addresses) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func newK8sEndpoint(endpoint Endpoint, k8sClient kubernetes.Interface) (*apiv1.Endpoints, error) {
```

Comment:
Nitpick. IMO this should be a method of the resource. Then we do not need to pass the client and also can do logging if we need to.

Reply:
Done.